src/main/java/emissary/test/core/ExtractionTest.java

```diff
@@ -155,10 +153,44 @@ protected void checkAnswers(Document answers, IBaseDataObject payload, List<IBas
     }
 
     protected void checkAnswers(Element el, IBaseDataObject payload, List<IBaseDataObject> attachments, String tname) throws DataConversionException {
+        // check to see if payload has extracted records and see if attachments is not null
+        // if true, implement check on payload and attachments to compare expected v found counts
+        // else, implement checkAttachmentAndExtractedRecordCount
+        if ((attachments != null) || payload.hasExtractedRecords()) {
```

Comment:
One case that does not work is if the <numAttachments>0</numAttachments> and you add <att1>...</att1>, this test still passes.

Reply:
Fixed the issue in the newest commit. Test now fails when numAttachments=0 and if att1 (or more) is present.

Reply:
Same issue with extracted records.

Reply:
If I don't define numAttachments and/or extractCount, tests pass.